### PR TITLE
Template values

### PR DESCRIPTION
**Pull request by mgu**
Cf. #65

Alternative and more flexible solution to resolve indirections

### Check list:
- [x] Write unit tests (100% coverage ?)
- [x] Write or edit integration tests, if applicable ?
- [x] Add a line in CHANGELOG.md

### CODE REVIEW

**Comment by ewjoachim** [comment]
(For the record, we had the idea together and paired a good part of it, so while I +1 and will review the PR, a real exterior eye is welcome)

**Comment by mgu** [comment]
rebased, ready for reviews :)

File: tests/unit/test_client_base.py [hunk 1]
```diff
@@ -351,3 +337,61 @@ def test_vault_client_move_secrets_overwrite_force(vault):
...
+    "vault_contents, expected",
+    [
+        # Secret is not a template
+        ({"a": "b"}, "b"),
+        # Secret not a string
+        ({"a": ["yay"]}, ["yay"]),
+        # Secret is a template without variable expansion
+        ({"a": "!template!b", "b": "c"}, "b"),
+        # Secret is a template
+        ({"a": "!template!{{ vault('b') }}", "b": "c"}, "c"),
```
Comment (SwannCroiset) [inline tests/unit/test_client_base.py]: what happens when `"b":  {"u": "x", "p": "z"}` ?
is it possible to access "u" and "p" values (ie. `"!template!{{ vault('b').u }}"` ?

Comment (mgu) [reply tests/unit/test_client_base.py]: here is an example :
```
$ vault set --yaml --stdin composite
username: foo
password: bar
Done
$ vault get composite
---
password: bar
username: foo
$ vault set username '!template!username={{ vault("composite")["username"] }}'         
Done
$ vault get --text username
username=foo
```

Comment (SwannCroiset) [reply tests/unit/test_client_base.py]: ok nice then

Comment (ewjoachim) [reply tests/unit/test_client_base.py]: Can we test this then ?

**Comment by ewjoachim** [comment]
Ah can you do something for the coverage if you haven't ?

**Comment by mgu** [comment]
coverage has labready been fixed

**Pull request merged**